dlt/common/destination/reference.py

```diff
@@ -64,13 +64,23 @@ class StorageSchemaInfo(NamedTuple):
     schema: str
 
 
-class StateInfo(NamedTuple):
+@dataclasses.dataclass
+class StateInfo:
```

Comment:
this is really cool, I was going to do this for the pluggable state destination anyway.